### ISSUE TITLE
No way to retrieve http status text from a `Response`

### ISSUE THREAD

**Issue by simonwuelker** [hyperium/http#737]
There is currently no way to get the status text (The `OK` of `HTTP/1.1 200 OK`) of a http `Response` object.
[`StatusCode::canonical_reason`](https://docs.rs/http/latest/http/status/struct.StatusCode.html#method.canonical_reason) only returns a standardized reason instead of the reason that was actually sent. 

Of course, the status text is equivalent to the canonical reason 99% of the time. However, there is no requirement that this is the case and it is required to implement things like javascript's [`XmlHttpRequest::statusText`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/statusText).

This is causing issues for servo in some web platform tests, like [`xhr/send-conditional.htm`](https://wpt.fyi/results/xhr/send-conditional.htm?label=master&label=experimental&product=servo), which encode the test result in the response status text.

**Comment by seanmonstar** [comment]
That's true, and it was a deliberate design decision for this crate. Reason phrases are dead in HTTP/2 and 3 and the future.

However, if you're using hyper, you can use this extension to grab it when it differs from the canonical reason: https://docs.rs/hyper/latest/hyper/ext/struct.ReasonPhrase.html

**Comment by simonwuelker** [comment]
Thank you for the quick response!
